File: Python-Basics/loops/cake.py
cake_length = int(input())
cake_width = int(input())

pieces_count = 0
cake_size = cake_length * cake_width

while cake_size > 0:
    pieces = input()
    if pieces == 'STOP':
        print(f'{cake_size} pieces are left.')
        break
    else:
        cake_size -= int(pieces)
        pieces_count += int(pieces)

if cake_size <= 0:
    print(f'No more cake left! You need {pieces_count - (cake_width*cake_length)} pieces more.')

File: CPP-Basics/Cake.cpp
#include <iostream>
#include <string>

using namespace std;

int main () {

    int w, l;
    cin >> w >> l;

    int cakeSize = w * l;

    string command;
    cin >> command;

    int pieces = 0;

    while(command != "STOP") {

        pieces += stoi(command);
        cakeSize -= pieces;
        pieces = 0;

        if(cakeSize < 0){

            cakeSize = (cakeSize - pieces) * -1;
            cout << "No more cake left! You need " << cakeSize << " pieces more." << endl;
            break;
        }

        cin >> command;

    }

    if(command == "STOP") {

    int left = cakeSize - pieces;

    cout << left << " pieces are left." << endl;

    }

    return 0;
}
